### ISSUE TITLE
Why tensorpack's tutorial FasterRCNN use channels first ?

### ISSUE THREAD

**Issue by cocopambag** [tensorpack/tensorpack#1268]
Hi, ppwwyyxx.

Well, analyzing your FasterRCNN's example, It seems to have found a small problem.
then, at tensorpack/examples/FasterRCNN/modeling/backbone.py line 210.

I saw tensorpack api document and tensorflow api document too.

Following is tensorpack api
`models.MaxPooling(variable_scope_name, inputs, pool_size, strides=None, padding='valid', data_format='channels_last')`

>Same as tf.layers.MaxPooling2D. Default strides is equal to pool_size.

and following is tensorflow api
`tf.layers.MaxPooling2D( pool_size, strides, padding='valid', data_format='channels_last', name=None, **kwargs
)`

>data_format: A string. The ordering of the dimensions in the inputs. channels_last (default) and channels_first are supported. **channels_last** corresponds to inputs with **_shape (batch, height, width, channels)_** while **channels_first** corresponds to inputs with **_shape (batch, channels, height, width)_**.

so.... at tensorpack/examples/FasterRCNN/modeling/backbone.py line 207.
`l.set_shape([None, chan, None, None])`
Use layer.set_shape() function. so, data format is channels_first !

on the other words, i think at tensorpack/examples/FasterRCNN/modeling/backbone.py line 207, 
`l = MaxPooling('pool0', l, 3, strides=2, padding='VALID')` should be changed to 
`l = MaxPooling('pool0', l, 3, strides=2, padding='VALID', data_format='channels_first')`

Isn't that right? 

Well, if this is true, Conv2d should also be changed

Thank you for reading my issue.

**Comment by cocopambag** [comment]
Well,  if this is true, Conv2d should also be changed

**Comment by ppwwyyxx** [comment]
The code already set them to use channels_first, at https://github.com/tensorpack/tensorpack/blob/610a8b9baed5ff8f9e8624538141b414f8578d4a/examples/FasterRCNN/modeling/backbone.py#L75

**Comment by cocopambag** [comment]
Aha, I see. 
Thank you for you comment.
